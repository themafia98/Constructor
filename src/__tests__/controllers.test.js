import React from 'react';
import renderer from 'react-test-renderer';
import Controllers from '../components/controllers/controllers';
import Events from 'events';

    let stream = new Events();

    const ControllersComponent = renderer.create(
        <Controllers
        targetSection = 'Header'
        menuActive = {true}
        eventStreamBuild  = {stream}
        countSection = {0}
        countComponents = {0}
        sizeParentBox = {{width: 1000, height: 1000, left: 0, top: 60}}
        />
    );

    test("Controllers test", () =>{

        let snapshotControllersComponent = ControllersComponent.toJSON();
        expect(snapshotControllersComponent).toMatchSnapshot();

        const addButton = ControllersComponent.root.find(item => item.props.className === 'ControllersEditComponent__addButton');
        addButton.props.onClick({stopPropagation: () => null});

        snapshotControllersComponent = ControllersComponent.toJSON();
        expect(snapshotControllersComponent).toMatchSnapshot();

        const imageToolC = ControllersComponent.root.find(item => item.props.className === 'ImageTool CompoentnsMenu_button');
        imageToolC.props.onClick({stopPropagation: () => null});

        snapshotControllersComponent = ControllersComponent.toJSON();
        expect(snapshotControllersComponent).toMatchSnapshot();

        const textTool = ControllersComponent.root.find(item => item.props.className === 'TextTool CompoentnsMenu_button');
        textTool.props.onClick({stopPropagation: () => null});

        snapshotControllersComponent = ControllersComponent.toJSON();
        expect(snapshotControllersComponent).toMatchSnapshot();

        const buttonTool = ControllersComponent.root.find(item => item.props.className === 'ButtonTool CompoentnsMenu_button');
        buttonTool.props.onClick({stopPropagation: () => null});

        snapshotControllersComponent = ControllersComponent.toJSON();
        expect(snapshotControllersComponent).toMatchSnapshot();

        const videoTool = ControllersComponent.root.find(item => item.props.className === 'VideoTool CompoentnsMenu_button');
        videoTool.props.onClick({stopPropagation: () => null});

        snapshotControllersComponent = ControllersComponent.toJSON();
        expect(snapshotControllersComponent).toMatchSnapshot();

    });