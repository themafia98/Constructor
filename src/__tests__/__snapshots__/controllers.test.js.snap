// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controllers test 1`] = `
<div
  className="ControllersEditComponent"
>
  <img
    alt="icon"
    className="ControllersEditComponent__addButton"
    draggable={false}
    onClick={[Function]}
    src="/img/addButton.png"
    title={null}
  />
</div>
`;

exports[`Controllers test 2`] = `
<div
  className="ControllersEditComponent"
>
  <img
    alt="icon"
    className="ControllersEditComponent__addButton"
    draggable={false}
    onClick={[Function]}
    src="/img/addButton.png"
    title={null}
  />
  <div
    className="ComponentsMenu"
  >
    <button
      className="ImageTool CompoentnsMenu_button"
      onClick={[Function]}
    >
      <span>
        Image
      </span>
    </button>
    <button
      className="TextTool CompoentnsMenu_button"
      onClick={[Function]}
    >
      <span>
        Text
      </span>
    </button>
    <button
      className="ButtonTool CompoentnsMenu_button"
      disabled={false}
      onClick={[Function]}
    >
      <span>
        Button
      </span>
    </button>
    <button
      className="VideoTool CompoentnsMenu_button"
      onClick={[Function]}
    >
      <span>
        Video
      </span>
    </button>
  </div>
</div>
`;

exports[`Controllers test 3`] = `
<div
  className="ControllersEditComponent"
>
  <img
    alt="icon"
    className="ControllersEditComponent__addButton"
    draggable={false}
    onClick={[Function]}
    src="/img/addButton.png"
    title={null}
  />
  <div
    className="ComponentsMenu"
  >
    <button
      className="ImageTool CompoentnsMenu_button"
      onClick={[Function]}
    >
      <span>
        Image
      </span>
    </button>
    <button
      className="TextTool CompoentnsMenu_button"
      onClick={[Function]}
    >
      <span>
        Text
      </span>
    </button>
    <button
      className="ButtonTool CompoentnsMenu_button"
      disabled={false}
      onClick={[Function]}
    >
      <span>
        Button
      </span>
    </button>
    <button
      className="VideoTool CompoentnsMenu_button"
      onClick={[Function]}
    >
      <span>
        Video
      </span>
    </button>
  </div>
</div>
`;

exports[`Controllers test 4`] = `
<div
  className="ControllersEditComponent"
>
  <img
    alt="icon"
    className="ControllersEditComponent__addButton"
    draggable={false}
    onClick={[Function]}
    src="/img/addButton.png"
    title={null}
  />
  <div
    className="ComponentsMenu"
  >
    <button
      className="ImageTool CompoentnsMenu_button"
      onClick={[Function]}
    >
      <span>
        Image
      </span>
    </button>
    <button
      className="TextTool CompoentnsMenu_button"
      onClick={[Function]}
    >
      <span>
        Text
      </span>
    </button>
    <button
      className="ButtonTool CompoentnsMenu_button"
      disabled={false}
      onClick={[Function]}
    >
      <span>
        Button
      </span>
    </button>
    <button
      className="VideoTool CompoentnsMenu_button"
      onClick={[Function]}
    >
      <span>
        Video
      </span>
    </button>
  </div>
</div>
`;

exports[`Controllers test 5`] = `
<div
  className="ControllersEditComponent"
>
  <img
    alt="icon"
    className="ControllersEditComponent__addButton"
    draggable={false}
    onClick={[Function]}
    src="/img/addButton.png"
    title={null}
  />
  <div
    className="ComponentsMenu"
  >
    <button
      className="ImageTool CompoentnsMenu_button"
      onClick={[Function]}
    >
      <span>
        Image
      </span>
    </button>
    <button
      className="TextTool CompoentnsMenu_button"
      onClick={[Function]}
    >
      <span>
        Text
      </span>
    </button>
    <button
      className="ButtonTool CompoentnsMenu_button"
      disabled={false}
      onClick={[Function]}
    >
      <span>
        Button
      </span>
    </button>
    <button
      className="VideoTool CompoentnsMenu_button"
      onClick={[Function]}
    >
      <span>
        Video
      </span>
    </button>
  </div>
</div>
`;

exports[`Controllers test 6`] = `
<div
  className="ControllersEditComponent"
>
  <img
    alt="icon"
    className="ControllersEditComponent__addButton"
    draggable={false}
    onClick={[Function]}
    src="/img/addButton.png"
    title={null}
  />
  <div
    className="ComponentsMenu"
  >
    <button
      className="ImageTool CompoentnsMenu_button"
      onClick={[Function]}
    >
      <span>
        Image
      </span>
    </button>
    <button
      className="TextTool CompoentnsMenu_button"
      onClick={[Function]}
    >
      <span>
        Text
      </span>
    </button>
    <button
      className="ButtonTool CompoentnsMenu_button"
      disabled={false}
      onClick={[Function]}
    >
      <span>
        Button
      </span>
    </button>
    <button
      className="VideoTool CompoentnsMenu_button"
      onClick={[Function]}
    >
      <span>
        Video
      </span>
    </button>
  </div>
</div>
`;
